### PR TITLE
[front] feat: add default unsafe preference for recommendations page

### PR DESCRIPTION
**Pull request by ShinseiNova**
**related to #1508**

### CODE REVIEW

**Comment by GresilleSiffle** [comment]
Here is the expected behaviour:

If a user (anonymous or authenticated) clicks on a specific recommendation URL containing recommendation filters, his/her personal preferences should not override those filters. This allow the users to share links between each other. This is the current behaviour.

If an anonymous user clicks on the recommendation link of the front end's side bar, the recommendation should be filtered using the poll's default recommendation filters. This is the current behaviour.

If an authenticated user clicks on the recommendation link of the front end's side bar:
- the poll's default filters should be used to initialize the filters
- the poll's default filters should be overridden by configured user settings (configured means != than undefined/null)

Example:

```bash
# considering those poll default filters
date=month&unsafe=false

# and the following user setting
unsafe=true

# the expected recommendation link in the sidebar is
https://tournesol.app/recommendations/?date=month&unsafe=true

# and not
https://tournesol.app/recommendations/?unsafe=true
```

**Review by GresilleSiffle — CHANGES_REQUESTED** [review]
Thank you @ShinseiNova 

Here is a first batch of improvements we can already work on. Let me know if you have any question.

**Review by GresilleSiffle — COMMENTED** [review]
The code is almost ready!

It could be good to have some functions that do the matching between the user's settings and the values expected by the front end. Currently it's the responsibility of the side bar, and in my opinion it should be moved in external functions for a greater readability / maintainability.

For instance the value `ALL_TIME` of the setting `recommendations__default_date` is translated in an empty string by the side bar. The boolean value `false` or the setting `recommendations__default_unsafe` is also translated in an empty string.

**Review by GresilleSiffle — COMMENTED** [review]
I removed the default languages setting for now.

As discussed with the rest of the team we don't want to maintain at the same time a preference system based on the local storage only (updating the lang from the recommendations page) and  a system based on the API (updating the lang from the preferences page).

We will first update the extension to allow the users to select their preferred languages, then we will re-introduced the default languages setting.

File: frontend/public/locales/en/translation.json [hunk 1]
```diff
@@ -328,8 +328,15 @@
     "preferencesUpdatedSuccessfully": "Preferences updated successfully.",
     "comparisonPage": "Comparison page",
     "rateLater": "Rate-later list",
+    "recommendationsPage": "Recommendations page",
     "updatePreferences": "Update preferences"
   },
+  "videosUserSettingsForm": {
+    "recommendations": {
+      "defaultUploadDate": "Default value of the Uploaded filter",
```
Comment (lfaucon) [inline frontend/public/locales/en/translation.json:336]:
```suggestion
      "defaultUploadDate": "Default value of the uploaded date filter",
```

File: frontend/public/locales/fr/translation.json [hunk 2]
```diff
@@ -336,8 +336,15 @@
     "preferencesUpdatedSuccessfully": "Préférences mises à jour avec succès.",
     "comparisonPage": "Comparaison (page)",
     "rateLater": "Liste à comparer plus tard",
+    "recommendationsPage": "Recommandations (page)",
     "updatePreferences": "Mettre à jour les préférences"
   },
+  "videosUserSettingsForm": {
+    "recommendations": {
+      "defaultUploadDate": "Valeur par défaut du filtre Mis en ligne",
```
Comment (lfaucon) [inline frontend/public/locales/fr/translation.json:344]:
```suggestion
      "defaultUploadDate": "Valeur par défaut du filtre date de mise en ligne",
```

File: frontend/src/features/settings/preferences/VideosPollUserSettingsForm.tsx [hunk 3]
```diff
@@ -43,6 +48,16 @@ const VideosPollUserSettingsForm = () => {
     DEFAULT_RATE_LATER_AUTO_REMOVAL
   );
 
+  // Recommendations (page)
+  const [recoDefaultUnsafe, setRecoDefaultUnsafe] = useState(false);
```
Comment (lfaucon) [inline frontend/src/features/settings/preferences/VideosPollUserSettingsForm.tsx:52]: When reloading the page, after having set my configuration to True, I see it showing false for a second, then showing my selected value. I'd recommend not displaying until the settings are fetched.

**Review by lfaucon — APPROVED** [review]

File: frontend/src/features/settings/preferences/VideosPollUserSettingsForm.tsx (hunk 3 above)
Comment (GresilleSiffle) [reply frontend/src/features/settings/preferences/VideosPollUserSettingsForm.tsx:54]: Fixed by https://github.com/tournesol-app/tournesol/pull/1553/commits/dab7cc70653f759fe23a1ad83c727faf74efbb68

**Pull request merged**